src/concurrency/local_epoch.cpp

```diff
@@ -16,17 +16,29 @@
 namespace peloton {
 namespace concurrency {
 
-  bool LocalEpoch::EnterEpoch(const uint64_t epoch_id) {
+  bool LocalEpoch::EnterEpoch(const uint64_t epoch_id, const bool is_snapshot_read) {
+
     epoch_lock_.Lock();
+
     // if this thread is never used or GC'd
     if (epoch_lower_bound_ == UINT64_MAX) {
+
       epoch_lower_bound_ = epoch_id - 1;
+    
     } else if (epoch_lower_bound_ >= epoch_id) {
-    // epoch_lower_bound_ has already been updated by the GC.
-    // have to grab a newer epoch_id.
-      epoch_lock_.Unlock();
-      return false;
-    } 
+    
+      if (is_snapshot_read == true) {
+    
+        epoch_lower_bound_ = epoch_id - 1;
```

Comment:
Why do we set the epoch_lower_bound_ to a smaller epoch? I think it's supposed to be a non-decreasing number, otherwise the GC process will be confused.